### PR TITLE
replace deprecated Queue and RateLimiter

### PR DESCRIPTION
**Pull request by Madhu-1**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug

 /kind cleanup

> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

Similar changes is done in external-resizer https://github.com/kubernetes-csi/external-resizer/pull/455

**Does this PR introduce a user-facing change?**:

```release-note
None
```

### CODE REVIEW

**Comment by leonardoce** [comment]
I run the smoke tests here, too, for 30 minutes.
Everything looks fine to me.

/lgtm

File: pkg/common-controller/snapshot_controller_base.go [hunk 1]
```diff
@@ -322,11 +330,11 @@ func (ctrl *csiSnapshotCommonController) snapshotWorker() {
 	}
 	defer ctrl.snapshotQueue.Done(keyObj)
 
-	if err := ctrl.syncSnapshotByKey(keyObj.(string)); err != nil {
+	if err := ctrl.syncSnapshotByKey(keyObj); err != nil {
```
Comment (jsafrane) [inline pkg/common-controller/snapshot_controller_base.go:333]: It's no longer `keyObj`, but directly `key`. Please update it in all workers.

Comment (Madhu-1) [reply pkg/common-controller/snapshot_controller_base.go]: @jsafrane done, PTAL

**Pull request merged**